airflow/www/auth.py

```diff
@@ -35,16 +44,100 @@ def get_access_denied_message():
     return conf.get("webserver", "access_denied_message")
 
 
-def has_access(permissions: Sequence[tuple[str, str]] | None = None) -> Callable[[T], T]:
```

Comment:
@vincbeck This is a breaking change for us, we have a plugin that uses this method. Can we please keep this method and work around it?

Reply:
Sadly this is really impossible to keep all the code in case someone uses that code in their own code (DAGs, plugins, ...). This goes back to definition of [public interface in Airflow](https://airflow.apache.org/docs/apache-airflow/stable/public-airflow-interface.html) and the security manager is not part of it. However, there is an option in Airflow where you can bring your own security manager using the constant `SECURITY_MANAGER_CLASS`, even though this is not part of the public interface we did no want to break user experience in case they are using a custom security manager so we handled it. But in your case, I feel like you need to update your code.

Reply:
`has_acess` has been in use by plugins since Airflow 1.10 - way before we had any formal definition of what read public or not.

In short: this is a breaking change and needs fixing

Reply:
+1, in our plugins we also use `airflow.www.auth.has_access` - if there is any alternative we could migrate before but maybe there are a lot of other plugins and users "in the wild" relying on this. So if the function at least is kept as a proxy to the new implementation it would be good. I also do not know any other alternative for the current main version how to check authentication for plugins if not using this method.